sound/soc/sof/ipc4-topology.c

```diff
@@ -1348,7 +1348,13 @@ static void sof_ipc4_unprepare_copier_module(struct snd_sof_widget *swidget)
 
 	if (WIDGET_IS_AIF(swidget->id) || swidget->id == snd_soc_dapm_buffer) {
 		if (pipeline->use_chain_dma) {
-			pipeline->msg.primary = 0;
+			/*
+			 * Preserve the DMA Link ID and clear other bits since
+			 * the DMA Link ID is only configured once during
+			 * dai_config, other fields are expected to be 0 for
+			 * re-configuration
+			 */
+			pipeline->msg.primary &= SOF_IPC4_GLB_CHAIN_DMA_LINK_ID_MASK;
```

Comment:
is this one correct? It's the host-facing part, in theory this has nothing to do with the link DMA.

Reply:
right, true, I'll drop this change

Reply:
done

Reply:
hmm, is this really dropping or should we used the dual macro SOF_IPC4_GLB_CHAIN_DMA_HOST_ID?

Reply:
The HOST ID is reprogrammed in `sof_ipc4_prepare_copier_module()`
We need to preserve the LinkID, all others must be cleared.

Reply:
ok